main/com.spotify/index.js

```diff
@@ -71,6 +71,11 @@ module.exports = class SpotifyDevice extends Tp.BaseDevice {
                 device_name: this.state.id,
                 token: this.accessToken
             });
+
+            this.on('state-changed', () => {
```

Comment:
You can override updateState() instead of connecting to state-changed, it's a bit cleaner wrt object lifetimes I think.
(reference for BaseDevice class: https://github.com/stanford-oval/thingpedia-api/blob/master/lib/base_device.ts#L122)

Reply:
The update of the device state accessToken happens in updateOAuth2Token (base_device.ts#L387) then the state-changed event is emitted.
In this case the updateState will not be fired because the structure is updated internally.

Reply:
Yeah I confused myself with the different state update methods. I suppose then `updateOAuth2Token` is the method to override...